scrapy/downloadermiddlewares/auth.py

```diff
@@ -28,19 +47,33 @@ def spider_opened(self, spider):
             self.auth = basic_auth_header(usr, pwd)
 
     def process_request(self, request, spider):
-        auth = getattr(self, 'auth', None)
-        if auth and 'Authorization' not in request.headers:
-            request.headers['Authorization'] = auth
```

Comment:
Why are you deleting these lines? Does your change not break http attributes given in the spider?